### PR TITLE
Use "aws_access_key_id" and "aws_secret_access_key" for reads, not just unload

### PR DESCRIPTION
**Pull request by koeninger**

### CODE REVIEW

File: src/main/scala/com/databricks/spark/redshift/RedshiftRelation.scala [hunk 1]
```diff
@@ -107,7 +108,7 @@ case class RedshiftRelation(jdbcWrapper: JDBCWrapper, params: MergedParameters,
     val filterClauses = filters map {
       case EqualTo(attr, value) => s"${sqlQuote(attr)} = ${compileValue(value)}"
       case LessThan(attr, value) => s"${sqlQuote(attr)} < ${compileValue(value)}"
-      case GreaterThan(attr, value) => s"${sqlQuote(attr)}) > ${compileValue(value)}"
```
Comment (JoshRosen) [inline src/main/scala/com/databricks/spark/redshift/RedshiftRelation.scala]: Another test coverage issue :(

Comment (koeninger) [inline src/main/scala/com/databricks/spark/redshift/RedshiftRelation.scala]: Yeah, there's a separate earlier PR just for that one line... got pulled into this branch by accident as I'm trying to get things working

**Comment by JoshRosen** [comment]
Hey @koeninger, do you have a regression test for this? Now that we have the infra from #41 in place I think it would be a good idea to test this.

**Comment by JoshRosen** [comment]
Hmm, I'm wondering why the fact that I'm missing this fix in my end-to-end tests has not led to a test failure yet.

**Comment by eduardoramirez** [comment]
@JoshRosen I came across this issue as well. It worked fine when running the job an EC2 instance that was allowed to read/write from S3. However, when running the job locally, I needed to set the hadoop configuration with my keys.

**Comment by JoshRosen** [comment]
@eduardoramirez @koeninger, given that both of you have reported this issue and fix I'm inclined to merge this now and figure out the regression tests later. I'm still puzzled over why I haven't been able to get this to fail in Travis, especially given that I purposely picked environment variables that don't match the ones that AWS expects. I'll take a closer look at our S3 bucket permissions tomorrow to see if that could explain the problem.

**Comment by JoshRosen** [comment]
I think I've spotted the problem: during test setup, we end up setting some Hadoop Configuration access key properties on the global `SparkContext.hadoopConfiguration`.  I'm going to remove those lines, verify that the tests fail, then will rebase and merge this fix.

**Comment by JoshRosen** [comment]
I've opened #55 to bring this up to date and to fix the tests so that they're capable of detecting the bug being fixed here.

**Pull request closed without merging**